src/Strings.f90

```diff
@@ -241,6 +247,48 @@ SUBROUTINE clear_str(this)
 ENDSUBROUTINE clear_str
 !
 !-------------------------------------------------------------------------------
+!> @brief breaks a string into partitions at the first instance of a delimiter
+!> @param this the string being partitioned
+!> @param sep the delimter used for partitioning
+!>
+PURE FUNCTION partition(this,sep) RESULT(tokens)
+  CLASS(StringType),INTENT(IN) :: this
+  CHARACTER(LEN=*),INTENT(IN) :: sep
+  TYPE(StringType) :: tokens(3)
```

Comment:
How does this behave if you assign the return value to an unallocated allocatable array of string types?  It might be good to make the return value allocatable instead, but I'm not sure if it's actually necessary.

Reply:
It seems to work in the unit test, which know is not the strongest endorsement, but I think it should be fine...
```
Note that even if A is an allocatable array, A(:) is not an allocatable array; it is an
array subsection.
integer, dimension(:), allocatable :: a
a = [ 1, 2, 3 ] ! Size of a becomes 3
a = [ 1, 2, 3, 4 ] ! Size of a becomes 4
a(:) = [ 1, 2, 3 ] ! Illegal because a(:) is size 4 and not allocatable
```

If a were a string type then when the function returns it should be a = [str1,str2,str3]. So the key is to not put `(:)` unless a is size 3.

Reply:
Ok, that's fair.  I'm still not used to relying on that because of old compilers and the fact that it has tripped me up when I've done it on accident, but I suppose that's how the language standard is written, and the compiler versions we use now tend to be fine with that implicit allocation.  So I'm good leaving it as is.